addons/base/views.py

```diff
@@ -532,6 +534,44 @@ def addon_delete_file_node(self, target, user, event_type, payload):
                 file_node.delete(user=user)
 
 
+def get_storage_usage_of_children(folder):
+    folder_children = folder.children.all()
+    total_storage = 0
+    for item in folder_children:
+        if item.kind == 'file':
+            total_storage += item.versions.all().aggregate(sum=Sum('size'))['sum']
+        elif item.kind == 'folder':
+            total_storage += get_storage_usage_of_children(item)
+    return total_storage
+
+
+@file_signals.file_updated.connect
+def storage_usage_cache_control(self, target, user, event_type, payload):
+    if storage_usage_flag_active():
+        provider = payload.get('provider')
+
+        if provider == 'osfstorage':
+            path = payload['metadata']['path']
+            osfstorage = target.get_addon('osfstorage')
+            storage_usage = cache.get('storage_usage:' + osfstorage._id)
+
+            if not storage_usage:
+                storage_usage = osfstorage.storage_usage
+            elif event_type in ('file_added', 'file_updated'):
+                file_node = OsfStorageFile.objects.get(_id=path)
+                storage_usage += file_node.versions.first().size
```

Comment:
Is `first()` guaranteed to return the latest version? Better to add an explicit `order_by` to make sure.

Reply:
Oh, I see, looks like we set `ordering` on `FileVersion`: https://github.com/CenterForOpenScience/osf.io/blob/ffc0b2cb22a7c08a51e878ccb04357d6f9977912/osf/models/files.py#L807-L808 .

So I guess this is fine. Never hurts to be explicit though.